server/src-lib/Hasura/RQL/DML/Internal.hs

```diff
@@ -247,11 +247,13 @@ validateHeaders depHeaders = do
     unless (hdr `elem` map T.toLower headers) $
     throw400 NotFound $ hdr <<> " header is expected but not found"
 
-simplifyError :: Q.PGTxErr -> Maybe T.Text
+simplifyError :: Q.PGTxErr -> Maybe (Code, T.Text)
 simplifyError txErr = do
   stmtErr <- Q.getPGStmtErr txErr
   codeMsg <- getPGCodeMsg stmtErr
-  extractError codeMsg
+  errTxt <- extractError codeMsg
```

Comment:
Let's change `extractError` itself to return both `code` and `message`. We'll probably need to use the correct codes for the rest of the Postgres errors too